src/main/java/org/jboss/tools/intellij/analytics/SaProcessExecutor.java

```diff
@@ -0,0 +1,142 @@
+	private static final Logger logger = Logger.getInstance(SaProcessExecutor.class);
+
+
+	/**
+	 * <p>Authenticate a CRDA CLI user.</p>
+	 *
+	 * If CLI config file is not present in host machine then user is considered to be new
+	 * and needs to be authenticated in CRDA Platform.
+	 *
+	 * @throws IOException In case of process failure
+	 * @throws InterruptedException In case of process failure
+	 */
+	public void authenticateUser() throws IOException, InterruptedException {
+		// Check if crda config file is already present in system. if not create CLI config file.
+		if(!Files.exists(Paths.get(SaConfig.CLI_CONFIG_FILE_PATH))) {
+			logger.info("Authenticating user.");
+
+			// Run CLI command to set user consent to False for CLI telemetry data collection.
+			String cmd = SaConfig.CLI_TELEMETRY_CONSENT;
+			if (!SystemInfo.isWindows) cmd = "./" + cmd;
+			execute(cmd);
+
+			// Run command to authenticate user in CRDA Platform.
+			cmd = SaConfig.CLI_SET_USER;
+			if (!SystemInfo.isWindows) cmd = "./" + cmd;
+			execute(cmd.replace("userID", UUID.randomUUID().toString()));
+		}
+	}
+
+
+	/**
+	 * <p>Perform Stack Analysis on given file.</p>
+	 *
+	 * @param filePath Path to target manifest file.
+	 *
+	 * @return String object having analysis report.
+	 *
+	 * @throws IOException In case of process failure
+	 * @throws InterruptedException In case of process failure
+	 */
+	public String performStackAnalysis(String filePath) throws IOException, InterruptedException {
+		logger.info("Starting Stack Analysis.");
+		// Authenticate user, in case file has been deleted after loading the plugin
+		authenticateUser();
+
+		String cmd = SaConfig.CLI_ANALYSE;
+		// Handle the space in directory names.
+		if (SystemInfo.isWindows) {
+			if (filePath.contains(" ")) filePath = "\"" + filePath + "\"";
+		} else {
+			filePath = filePath.replaceAll(" ", "\\\\ ");
+			cmd = "./" + cmd;
+		}
+
+		// Set the path of directory having CLI binary into command
+		cmd = cmd.replace("filePath", filePath);
+
+		// Execute CLI command for analysis
+		return execute(cmd);
+	}
+
+
+	private String getPrompt() {
+		if (SystemInfo.isWindows)
+			return "cmd.exe";
+		return "bash";
+	}
+	
+	private String getCmd() {
+		if (SystemInfo.isWindows)
+			return "/c";
+		return "-c";
+	}
+
+
+	/**
+	 * <p>Execute CLI commands.</p>
+	 *
+	 * @param command Command to be executed.
+	 *
+	 * @return String object having result of command.
+	 *
+	 * @throws IOException In case of process failure
+	 * @throws InterruptedException In case of process failure
+	 */
+	public String execute(String command) throws IOException, InterruptedException {
+		// Logic to execute given CLI command and get the result.
+		ProcessBuilder processBuilder = new ProcessBuilder();
+		// Set CLI binary location as working directory for process.
+		processBuilder.directory(new File(Platform.pluginDirectory));
+		processBuilder.command(this.getPrompt(), this.getCmd(), command);
```

Comment:
Why do you need to go through the command shell to launch your process ?

Reply:
I am using CRDA CLI binary to run SA in background, I found that ProcessBuilder would be a good fit, there are other options as well like Runtime.exec(). If there is any better approach for it, please suggest me I will surly look into that.

Reply:
Yes ProcessBuilder is ok but why don't you use it to run the CLI directly and not through the system shell (cmd.exe / sh)